### PR TITLE
Use GCP auth action

### LINKED ISSUE #791: GHA google cloud warning
I noticed this in the **Setup GCP** step of the binary builds.

>Warning: "service_account_key" has been deprecated. Please switch to using google-github-actions/auth which supports both Workload Identity Federation and Service Account Key JSON authentication. For more details, see https://github.com/google-github-actions/setup-gcloud#authorization

### PR DESCRIPTION
**Pull request by emlys**
# Description
Fixes #791 
The GCP steps don't run in PRs, so be sure to check them on my branch.

# Checklist
- [ ] ~Updated HISTORY.rst (if these changes are user-facing)~
- [ ] ~Updated the user's guide (if needed)~

### CODE REVIEW

File: .github/workflows/binary-applications.yml [hunk 1]
```diff
@@ -251,9 +261,7 @@ jobs:
             - name: Set up GCP
               # Secrets not available in PR so don't use GCP.
               if: github.event_name != 'pull_request'
-              uses: google-github-actions/setup-gcloud@master
-              with:
-                  service_account_key: ${{ secrets.GOOGLE_SERVICE_ACC_KEY }}
+              uses: google-github-actions/setup-gcloud@v0
```
Comment (phargogh) [inline .github/workflows/binary-applications.yml:264]: Do we not need to authenticate in this build job?

Comment (phargogh) [reply .github/workflows/binary-applications.yml:264]: Actually, looking into the [sampledata build job on your fork](https://github.com/emlys/invest/runs/4720464510?check_suite_focus=true), I see a bunch of errors sprinkled throughout:

```
ServiceException: 401 Anonymous caller does not have storage.objects.create access to the Google Cloud Storage object.
```

So maybe we do need to authenticate?

**Review by phargogh — CHANGES_REQUESTED** [review]
Thanks for updating this @emlys !  I think there's one place where we might also need to authenticate, but looks good otherwise!

File: .github/workflows/binary-applications.yml (hunk 1 above)
Comment (emlys) [reply .github/workflows/binary-applications.yml]: Oh, looks like I just missed that one!

**Review by phargogh — APPROVED** [review]
Good to go!  Thanks @emlys !

**Pull request merged**